### PR TITLE
PSM V2 integration

### PR DESCRIPTION
**Pull request by nthoma**

### CODE REVIEW

File: thalaprotocol_demo/sources/demo.move [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module thalaprotocol_demo::demo {
+    use aptos_framework::object::{Self, Object};
+    use aptos_framework::primary_fungible_store;
+    use aptos_framework::fungible_asset::{FungibleAsset, Metadata};
+
+    use thalaprotocol_interface::psm_v2::{Self, PSM};
+
+    // PSM V2 Methods
+
+    fun mint(
```
Comment (LawsonGraham) [inline thalaprotocol_demo/sources/demo.move]: comments on input and output would be great, i.e. is MOD in or out?

File: thalaprotocol_interface/sources/psm_v2.move [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        abort 0
+    }
+
+    #[view]
+    public fun redeem_paused(): bool {
+        abort 0
+    }
+
+    #[view]
+    public fun psm_rate_limiter_config(_psm_obj: Object<PSM>): (u64, u128, u64, u128) {
```
Comment (LawsonGraham) [inline thalaprotocol_interface/sources/psm_v2.move]: comment on what these values are would be good

File: thalaprotocol_interface/sources/psm_v2.move [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        abort 0
+    }
+
+    #[view]
+    public fun redeemable_mod(_psm_obj: Object<PSM>): u64 {
+        abort 0
+    }
+
+    #[view]
+    public fun redeemable_mod_rate_limited(_psm_obj: Object<PSM>): u64 {
```
Comment (LawsonGraham) [inline thalaprotocol_interface/sources/psm_v2.move]: comment instructing that this is preferred over `redeemable_mod`

**Review by LawsonGraham — APPROVED** [review]
generally lgtm, just prefer more comments to minimize questions from integrators

File: README.md [hunk 4]
```diff
@@ -12,6 +12,9 @@ ThalaSwapV2: https://explorer.aptoslabs.com/account/0x007730cd28ee1cdc9e999336cb
 
 ThalaSwapLens: https://explorer.aptoslabs.com/account/0xff1ac437457a839f7d07212d789b85dd77b3df00f59613fcba02388464bfcacb/transactions?network=mainnet
 
+TODO: Add ThalaProtocol
+ThalaProtocol: ...
```
Comment (0xbe1) [inline README.md]: here's the address 0x6f986d146e4a90b828d8c12c14b6f4e003fdff11a8eecceceb63744363eaac01

File: README.md [hunk 5]
```diff
@@ -12,6 +12,9 @@ ThalaSwapV2: https://explorer.aptoslabs.com/account/0x007730cd28ee1cdc9e999336cb
 
 ThalaSwapLens: https://explorer.aptoslabs.com/account/0xff1ac437457a839f7d07212d789b85dd77b3df00f59613fcba02388464bfcacb/transactions?network=mainnet
 
+TODO: Add ThalaProtocol
```
Comment (0xbe1) [inline README.md]: also plz update thalaswap v2 and thalaswap lens link to exclude "/transactions"

File: thalaprotocol_demo/Move.toml
Comment (0xbe1) [inline thalaprotocol_demo/Move.toml:1]: how about thala_protocol_demo

File: thalaprotocol_interface/Move.toml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+[package]
+name = "ThalaProtocolInterface"
+version = "1.0.0"
+
+[addresses]
+thalaprotocol_interface = "48271d39d0b05bd6efca2278f22277d6fcc375504f9839fd73f74ace240861af"
+thalaprotocol_interface_deployer = "1bf23f0881f8fa149500ff6b7a047f608967c028a8ad7a2100caa84833ce851d"
```
Comment (0xbe1) [inline thalaprotocol_interface/Move.toml]: dont need deployer

**Review by 0xbe1 — APPROVED** [review]
lgtm

**Review by 0xbe1 — APPROVED** [review]
lgtm

**Pull request merged**